### PR TITLE
8346174: UMAX/UMIN are missing from XXXVector::reductionOperations

### PR DESCRIPTION
**Pull request by PaulSandoz**
Add functional support for unsigned min/max reductions on vectors.

We also need to ensure that the `reductionCoerced` intrinsic bails out when there is no reduction operation for the lanewise operation. When intrinsic support is added for integral vectors this will still be the case for floating point vectors.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8346174](https://bugs.openjdk.org/browse/JDK-8346174): UMAX/UMIN are missing from XXXVector::reductionOperations (**Bug** - P3)

### Reviewers
 * [Quan Anh Mai](https://openjdk.org/census#qamai) (@merykitty - Committer)
 * [Jatin Bhateja](https://openjdk.org/census#jbhateja) (@jatin-bhateja - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22741/head:pull/22741` \
`$ git checkout pull/22741`

Update a local copy of the PR: \
`$ git checkout pull/22741` \
`$ git pull https://git.openjdk.org/jdk.git pull/22741/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22741`

View PR using the GUI difftool: \
`$ git pr show -t 22741`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22741.diff">https://git.openjdk.org/jdk/pull/22741.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22741#issuecomment-2542048260)
</details>

### CODE REVIEW

File: src/hotspot/share/opto/vectorIntrinsics.cpp [hunk 1]
```diff
@@ -1385,6 +1385,12 @@ bool LibraryCallKit::inline_vector_reduction() {
   int num_elem = vlen->get_con();
   int opc  = VectorSupport::vop2ideal(opr->get_con(), elem_bt);
   int sopc = ReductionNode::opcode(opc, elem_bt);
+  if (sopc == opc) {
```
Comment (merykitty) [inline src/hotspot/share/opto/vectorIntrinsics.cpp:1388]: Can you merge this with the following `if`?

**Review by merykitty — APPROVED** [review]
LGTM otherwise.

**Comment by merykitty** [comment]
@jatin-bhateja You may want to implement reduction intrinsics for these operations.

File: src/hotspot/share/opto/vectorIntrinsics.cpp (hunk 1 above)
Comment (PaulSandoz) [reply src/hotspot/share/opto/vectorIntrinsics.cpp]: Yes, i merged them.

**Review by merykitty — APPROVED** [review]

**Comment by jatin-bhateja** [comment]
> @jatin-bhateja You may want to implement reduction intrinsics for these operations.

Thanks @merykitty assigned this to myself
https://bugs.openjdk.org/browse/JDK-8346256

**Pull request closed without merging**